### PR TITLE
Exit early on setup-deps

### PR DESCRIPTION
**Pull request by pecigonzalo**
We had CI runs fail on a command and continue
making debugging the root-cause hard.
eg. https://buildkite.com/sourcegraph/qa/builds/997#f0836819-a628-4dca-a5b1-21f96001c3cd/63-178

This makes scripts fail early and run in "strict mode".

### CODE REVIEW

**Comment by pecigonzalo** [comment]
This might make `yes | gcloud auth configure-docker` fail, checking now.

**Comment by pecigonzalo** [comment]
The `echo yes` should make it work.

**Review by uwedeportivo — APPROVED** [review]

**Review by daxmc99 — APPROVED** [review]
LGTM
This file seems like a prime candidate to be optimized though. This script is called in quite a few places where it performs identical tasks. 

Could we add this to the base CI docker image instead?

**Comment by pecigonzalo** [comment]
@daxmc99 , @davejrt suggested the same, we thought it was a good idea.

TLDR: 
In general I prefer CI runs installing their deps or using a specific image that does that for them. The benefit of this is that the pipeline has to define all deps and they are not just something you have to infer.

Given these runs are all the same, they basically do use a specific image just for them. If we need a generic bare-metal runner, we can create a new "bare" image which has no deps pre-installed.

**Comment by daxmc99** [comment]
> they basically do use a specific image just for them
This script takes about 15 seconds on CI. If we did bake that into an image we would get faster builds across a few places.

**Comment by pecigonzalo** [comment]
@daxmc99 I dont know if you can completely trim it. The `apt` part is not removed, as it was just for installing `crcmod` which is not required anymore and `asdf` you cant bake the exact version.
We could cache a few of those on the image so it does not download them.
